packages/core/src/components/region/RegionModal/RegionModal.tsx

```diff
@@ -113,8 +124,10 @@ function RegionModal({
             setInput(e.currentTarget.value)
           }}
           onSubmit={handleSubmit}
-          fadeOutOnSubmit={true}
-          onClear={() => setInput('')}
+          fadeOutOnSubmit={false}
+          onClear={resetInputField}
+          inputButtonActionText={loading ? '...' : 'Apply'}
+          dismissible={!isMandatory}
```

Comment:
If you apply @lariciamota suggestion, just a reminder to change here too 🎗️ 

```suggestion
          dismissible={isDismissible}
```